threestudio/systems/zero123.py

```diff
@@ -45,35 +46,45 @@ def on_fit_start(self) -> None:
             ],
         )
 
-    def training_step(self, batch, batch_idx):
+    def log_loss(self, name, loss_term, weight=1.0):
+        self.log(name, loss_term)
+        loss_term_weighted = weight * loss_term
+        self.log(name + "_w", loss_term_weighted)
+        return loss_term_weighted
+
+    def training_substep(self, batch, batch_idx, do_ref, loss_weight=1.0):
```

Comment:
TODO: generalize to support reference-image-supervision + DeepFloyd-guidance, etc.